lib/rolify/adapters/active_record/role_adapter.rb

```diff
@@ -105,14 +154,19 @@ def build_conditions(relation, args)
 
       def build_query(role, resource = nil)
         return [ "#{role_table}.name = ?", [ role ] ] if resource == :any
-        query = "((#{role_table}.name = ?) AND (#{role_table}.resource_type IS NULL) AND (#{role_table}.resource_id IS NULL))"
+        query = "((#{role_table}.name = ?) AND (#{role_table}.resource_type IS NULL) AND (#{role_table}.resource_id IS NULL) AND (#{role_table}.resource_uuid IS NULL))"
         values = [ role ]
         if resource
           query.insert(0, "(")
-          query += " OR ((#{role_table}.name = ?) AND (#{role_table}.resource_type = ?) AND (#{role_table}.resource_id IS NULL))"
+          query += " OR ((#{role_table}.name = ?) AND (#{role_table}.resource_type = ?) AND (#{role_table}.resource_id IS NULL) AND (#{role_table}.resource_uuid IS NULL))"
           values << role << (resource.is_a?(Class) ? resource.to_s : resource.class.name)
           if !resource.is_a? Class
-            query += " OR ((#{role_table}.name = ?) AND (#{role_table}.resource_type = ?) AND (#{role_table}.resource_id = ?))"
+            id_field = resource_id_field(resource)
+            if id_field == :resource_uuid
+              query += " OR ((#{role_table}.name = ?) AND (#{role_table}.resource_type = ?) AND (#{role_table}.resource_uuid = ?) AND (#{role_table}.resource_id IS NULL))"
+            else
+              query += " OR ((#{role_table}.name = ?) AND (#{role_table}.resource_type = ?) AND (#{role_table}.resource_id = ?) AND (#{role_table}.resource_uuid IS NULL))"
+            end
```

Comment:
Does it even matter what the value of the unimportant column is? We probably don't even need to make sure it's null. I'd imagine we can simplify some of the other logic where there's duplication like this similarly.

```suggestion
            query += " OR ((#{role_table}.name = ?) AND (#{role_table}.resource_type = ?) AND (#{role_table}.#{id_field} = ?))"
```